app/vmagent/main.go

```diff
@@ -315,10 +315,18 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 		return true
 	case "/opentelemetry/api/v1/push", "/opentelemetry/v1/metrics":
 		opentelemetryPushRequests.Inc()
-		if err := opentelemetry.InsertHandler(nil, r); err != nil {
+		if respBody, err := opentelemetry.InsertHandler(nil, r); err != nil {
 			opentelemetryPushErrors.Inc()
 			httpserver.Errorf(w, r, "%s", err)
 			return true
+		} else {
+			if respBody != nil {
+				w.Write(respBody)
+			}
+		}
+		ct := r.Header.Get("Content-Type")
+		if ct != "" {
+			w.Header().Set("Content-Type", ct)
```

Comment:
```go
respBody, err := opentelemetry.InsertHandler(nil, r)
	if err != nil {
		opentelemetryPushErrors.Inc()
		httpserver.Errorf(w, r, "%s", err)
		return true
	}
	
	if ct := r.Header.Get("Content-Type"); ct != "" {
		w.Header().Set("Content-Type", ct)
	}
	w.WriteHeader(http.StatusOK)
	if respBody != nil {
		w.Write(respBody)
	}
```

You need to call WriteStatus and Header CT before write because

// Write writes the data to the connection as part of an HTTP reply.
	//
	// If [ResponseWriter.WriteHeader] has not yet been called, Write calls
	// WriteHeader(http.StatusOK) before writing the data. If the Header
	// does not contain a Content-Type line, Write adds a Content-Type set
	// to the result of passing the initial 512 bytes of written data to
	Write([]byte) (int, error)

Reply:
fixed